src/panels/logbook/ha-logbook.ts

```diff
@@ -207,32 +207,42 @@ export class HaLogbook extends LitElement {
     );
   }
 
-  private _unsubscribe() {
+  private async _unsubscribe(): Promise<void> {
     if (this._subscribed) {
-      this._subscribed.then((unsub) => unsub?.());
-      this._subscribed = undefined;
+      try {
+        const unsub = await this._subscribed;
+        if (unsub) {
+          await unsub();
+          this._subscribed = undefined;
+          this._pendingStreamMessages = [];
+        }
+      } catch (err: any) {
+        // eslint-disable-next-line
+        console.error("Error unsubscribing:", err);
+      }
     }
   }
 
-  public connectedCallback() {
+  public async connectedCallback() {
     super.connectedCallback();
     if (this.hasUpdated) {
-      this._subscribeLogbookPeriod(this._calculateLogbookPeriod());
+      // Ensure clean state before subscribing
+      await this._subscribeLogbookPeriod(this._calculateLogbookPeriod());
```

Comment:
```suggestion
      this._subscribeLogbookPeriod(this._calculateLogbookPeriod());
```

it needs to run but we don't need to wait for this.